src/App.js

```diff
@@ -1,27 +1,73 @@
 import './App.css';
 import React from 'react';
 import SearchAndFilter from './components/SearchAndFilter';
+import jwt_decode from "jwt-decode";
 import 'bootstrap/dist/css/bootstrap.min.css';
+import {Button, Dropdown, DropdownButton}  from "react-bootstrap";
 
 class App extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      data: null
+      token: null,
+      user: {}
     };
   }
 
+  handleCallbackResponse = (response) => {
+    this.setState({token: response.credential});
+    let userObject = jwt_decode(response.credential);
+    this.setState({user: userObject});
+    document.getElementById("signInDiv").hidden = true;
+  }
+
+  handleLogout = () => {
+    this.setState({user: {}});
+    document.getElementById("signInDiv").hidden = false;
+  }
+
   componentDidMount() {
+    /* global google */
+    google.accounts.id.initialize({
+      client_id: process.env.REACT_APP_GOOGLE_CLIENT_ID,
+      callback: this.handleCallbackResponse
+    });
+
+    google.accounts.id.renderButton(
+      document.getElementById("signInDiv"),
+      { theme: "outline", size: "large"}
+    );
     
   }
 
   render() {
     return (
       <div>
         <SearchAndFilter />
+        {
+          Object.keys(this.state.user).length != 0 && (
+            <div style={styles.main}>
+              <DropdownButton title="My Account">
+                <Dropdown.Item as="button">Home</Dropdown.Item>
+                <Dropdown.Item as="button">My Recipes</Dropdown.Item>
+                <Dropdown.Item as="button">My Meal Plan</Dropdown.Item>
+                <Dropdown.Item as="button" onClick={this.handleLogout}>Sign out</Dropdown.Item>
+              </DropdownButton>
+            </div>
+          )
+        }
+        <div style={styles.main} id="signInDiv"></div>
```

Comment:
maybe there should be another MenuComponent that takes care of rendering whether there is Sign in vs Sign out button?

Reply:
I could do that...The current way I think is fine, but may be kind of messy. What do you say?

Reply:
then you can have <Menu /> rather than { .... Object.keys(...) }

If you put everything in App.js like this, then what's preventing you from putting everything in App.js? Then App.js becomes this conglomerate of things making it hard to test, read, and maintain.

Reply:
rather than move it into a separate component, I just moved that block of code into a function and called the function within the render.